ppci/lang/c/semantics.py

```diff
@@ -320,17 +320,21 @@ def check_redeclaration_storage_class(self, sym, declaration):
         old_storage_class = sym.declaration.storage_class
         new_storage_class = declaration.storage_class
         # None == automatic storage class.
-        invalid_combos = [(None, "static"), ("extern", "static")]
+        # changes of linkage between internal and external are illegal
+        invalid_combos = [(None, "static"),
+                          ("extern", "static"),
```

Comment:
I changed this code a bit, so that there are booleans checking for previously static or not.